tests/chainer_tests/functions_tests/math_tests/test_average.py

```diff
@@ -47,7 +47,8 @@ def setUp(self):
         # Sample weights. Weights should not sum to 0.
         while True:
             self.w = numpy.random.uniform(-2, 2, w_shape).astype(self.dtype)
-            if self.w.sum() > 1e-2:
+            w_sum_eps = 1.0 if self.dtype == numpy.float16 else 5e-2
+            if abs(self.w.sum()) > w_sum_eps:
```

Comment:
As discussed offline, could you try sum with high precision?

Reply:
I tried `abs(self.w.astype(numpy.float64).sum()) > w_sum_eps`, but the test failed.

Reply:
I see.